pkg/controllers/tunnel/tunnel.go

```diff
@@ -137,19 +133,23 @@ func (t *TunnelController) handleRemovedEndpoint(obj interface{}) {
 	if object, ok = obj.(*v1.Endpoint); !ok {
 		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
 		if !ok {
-			utilruntime.HandleError(fmt.Errorf("error decoding object, invalid type"))
-			klog.Errorf("problem decoding object")
+			utilruntime.HandleError(fmt.Errorf("Could not convert object %v to an Endpoint", obj))
 			return
 		}
 		object, ok = tombstone.Obj.(*v1.Endpoint)
 		if !ok {
-			utilruntime.HandleError(fmt.Errorf("error decoding object tombstone, invalid type"))
-			klog.Errorf("problem decoding object tombstone")
+			utilruntime.HandleError(fmt.Errorf("Could not convert object tombstone %v to an Endpoint", tombstone.Obj))
 			return
 		}
 		klog.V(4).Infof("Recovered deleted object '%s' from tombstone", object.GetName())
 	}
-	klog.V(4).Infof("Informed of removed endpoint for tunnel controller object: %v", object)
-	t.ce.RemoveCable(object.Spec.CableName)
-	klog.V(4).Infof("Removed endpoint from cable engine %s", object.Name)
+
+	klog.V(4).Infof("Informed of removed endpoint for tunnel controller object: %#v", object)
+	if err := t.ce.RemoveCable(object.Spec.CableName); err != nil {
+		utilruntime.HandleError(fmt.Errorf("error removing endpoint cable %s from engine: %v",
+			object.Spec.CableName, err))
+		return
+	}
+
+	klog.V(4).Infof("Removed endpoint cable %s from engine", object.Spec.CableName)
```

Comment:
true nitpicking: 
Can we keep the original info on that last log message (the cable engine) , although now we just have one?

klog.V(4).Infof("Removed endpoint cable %s from engine %s", object.Spec.CableName, object.Name)

Reply:
The previous wording was misleading, ie it implied it "removed a cable from an engine named X". Of course what we really mean is it "removed a cable named X from the engine".  Also since we're calling RemoveCable with 'object.Spec.CableName', I think it makes sense to substitute object.Spec.CableName for X.

Reply:
Oh, sorry, that's an endpoint, when I was making the comment I thought it was the cable engine name.., you're right...